app/ide-desktop/lib/dashboard/src/serviceWorker.ts

```diff
@@ -1,6 +1,17 @@
 /** @file A service worker that redirects paths without extensions to `/index.html`. */
 /// <reference lib="WebWorker" />
 
+// =================
+// === Constants ===
+// =================
+
+const IDE_CDN_URL = 'https://ensocdn.s3.us-west-1.amazonaws.com/ide'
+const FALLBACK_VERSION = '2023.1.1-nightly.2023.4.13'
```

Comment:
if you think it's a good idea to remove it though (since cloud frontend isn't the focus) that's easy enough too

Reply:
1. There is no documentation what `FALLBACK_VERSION` is, so I can't really tell what to do with it until I understand what it is and how it is meant to be used.
2. If something is cloud-release-specific, it should have that written in the name explicitly.
3. cloud release is not our focus, but we can't break it. We need to test it constantly and be sure it works.

Reply:
it's not possible to make opening the IDE work with cloud frontend because no `index.js.gz` has `runApp` (partly since it's not merged in yet). maybe we can do something like `(window.runApp ?? window.enso.main)` instead?

(or we can just leave it as-is and just stop supporting old versions, which to be fair isn't a bad idea since it won't be possible to create the old versions anyway)

Reply:
We should not pollute window scope with such names as `runApp`, as they are not explicit what app we are talking about. The previous name `window.enso.main` was much better. Can we leave it as it was, please?

Reply:
oh, that's been changed already for the desktop backend (`runApp` is no longer a global after the previous round of reviews).
about `window.enso.main`, i don't think that was ever intended to work - it runs the *entire* app, which includes the authentication flow on the desktop IDE. of course, this isn't a problem for cloud, but assuming `window.enso.main` is the way to go here, something else *must* used for the desktop backend

Reply:
so assuming `runApp` polluting the window is the main issue, the current solution i'd assume would be something like this:
```ts
const runApp = platform === platformModule.Platform.desktop ?
    appRunner.runApp :
    window.enso.main;
```
thoughts/suggestions?

Reply:
> isn't a problem for cloud, but assuming `window.enso.main` is the way to go here, something else _must_ used for the de

I don't understand these points. 
1. `window.enso.main` was intended to work and was working fine for long time.
2. What is the problem with authentication flow on the desktop version?
3. Your last sentence states "something else must used for the desktop backend" - why? I do not understand why we can't have one single entry point here. Inside of this entry point you are checking whether the app is run in cloud or not, so why we can't have just one entry point to start the app?

Reply:
> so assuming `runApp` polluting the window is the main issue, the current solution i'd assume would be something like this:
> 
> ```ts
> const runApp = platform === platformModule.Platform.desktop ?
>     appRunner.runApp :
>     window.enso.main;
> ```
> 
> thoughts/suggestions?

What is `appRunner`? Is it another global thing? Why can't we just have `window.enso.main` or `window.enso.run` (which would be even better than `main`).

Reply:
1. and it does still work, but not on desktop IDE (which is why i needed to change it in the first place)
2. sorry, bad wording - there is no problem with the authentication flow, the issue on the desktop IDE is that calling `window.enso.main` from the IDE React component runs the authentication flow instead of the actual IDE
3. `window.enso.main` is the entry point into the entire application, so i don't think the component that starts up the *IDE* should be running the *application* entry point

Reply:
`appRunner` is passed into `authentication.run()` (the React frontend which includes authentication and the dashboard), in order to avoid needing a global